src/posix.mak

```diff
@@ -22,9 +22,15 @@ INSTALL_DIR=../../install
 SYSCONFDIR=/etc
 PGO_DIR=$(abspath pgo)
 
-C=backend
-TK=tk
-ROOT=root
+D = ddmd
+
+C=$D/backend
+TK=$D/tk
+ROOT=$D/root
+
+GENERATED = ../generated
+G = $(GENERATED)/$(OS)$(MODEL)
+$(shell mkdir -p $G)
```

Comment:
> I think that there is no good reason to have two unused levels in the hierarchy

Well you can do `debug` and `release` builds at the same time without any issues ... and there are tons of other different build types available for DMD: https://github.com/dlang/dmd/blob/master/src/posix.mak#L160